app/cli/internal/action/attestation_init.go

```diff
@@ -186,3 +196,47 @@ func (action *AttestationInit) Run(ctx context.Context, opts *AttestationInitRun
 
 	return attestationID, nil
 }
+
+func enrichContractMaterials(ctx context.Context, schema *v1.CraftingSchema, client pb.AttestationServiceClient, logger *zerolog.Logger) error {
+	contractMaterials := schema.GetMaterials()
+	for _, pgAtt := range schema.GetPolicyGroups() {
+		group, _, err := policies.LoadPolicyGroup(ctx, pgAtt, &policies.LoadPolicyGroupOptions{
+			Client: client,
+			Logger: logger,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to load policy group: %w", err)
+		}
+		logger.Debug().Msgf("adding materials from policy group '%s'", group.GetMetadata().GetName())
+
+		toAdd := overrideMaterials(group, contractMaterials, logger)
+		contractMaterials = append(contractMaterials, toAdd...)
+	}
+
+	schema.Materials = contractMaterials
+
+	return nil
+}
+
+// override existing materials and return new ones
+func overrideMaterials(group *v1.PolicyGroup, fromContract []*v1.CraftingSchema_Material, logger *zerolog.Logger) []*v1.CraftingSchema_Material {
+	toAdd := make([]*v1.CraftingSchema_Material, 0)
+	for _, groupMaterial := range group.GetSpec().GetPolicies().GetMaterials() {
+		// check if material already exists in the contract, and override it
+		found := false
+		for i, mat := range fromContract {
+			if mat.GetName() == groupMaterial.GetName() {
```

Comment:
I personally think it should be the other way around. Contracts are more specific than groups. Groups are meant to be shared, but if you have a contract you could override. 

Any thoughts on why to do it the other way around?

Reply:
It was about enforcing the materials, but you are right, the other way around is way more flexible for users.